### PR TITLE
fix(cli): Remove --tag option from trigger delivery command, remove deprecated new-artifact command

### LINKED ISSUE #7368: CLI: appending image twice in the trigger delivery cmd
**Environment**

* Client OS (e.g., Linux, macOS, Windows):
* Keptn Version (`keptn version`): 0.14.1
* Kubernetes Cloud Provider (e.g., GKE, AKS):
* Kubernetes version (`kubectl version`):

**Affected Component**

* [x] CLI

**Describe the bug**

@grabnerandi discover a problem with the CLI and the trigger command. When the trigger delivery has only the `--image` flag with already the image tag, the tag is applied twice.

**Screenshots**

image (12)

**Additional context**

https://keptn.slack.com/archives/CNRCGFU3U/p1649152679410959

### PR DESCRIPTION
**Pull request by bacherfl**
Closes #7368

This PR removes the `--tag` option from the `trigger delivery` command in the Keptn CLI, since the `--image` option already supports passing the tag of a image via `--image=org/image:tag`

Also, this PR removes the `send new-artifact` command which has now been deprecated for over a year - this is a breaking change

integration test run: https://github.com/keptn/keptn/actions/runs/2102133825

### CODE REVIEW

File: cli/cmd/trigger_delivery.go [hunk 1]
```diff
@@ -170,39 +171,36 @@ func doTriggerDelivery(deliveryInputData deliveryStruct) error {
...
 	trimmedImage := strings.TrimSuffix(*deliveryInputData.Image, "/")
-	deliveryInputData.Image = &trimmedImage
 
-	if deliveryInputData.Tag == nil || *deliveryInputData.Tag == "" {
-		*deliveryInputData.Image, *deliveryInputData.Tag = docker.SplitImageName(*deliveryInputData.Image)
-	}
-	return docker.CheckImageAvailability(*deliveryInputData.Image, *deliveryInputData.Tag, nil)
+	image, tag := docker.SplitImageName(trimmedImage)
+	sanitizedImageName := fmt.Sprintf("%s:%s", image, tag)
+	deliveryInputData.Image = &sanitizedImageName
```
Comment (warber) [inline cli/cmd/trigger_delivery.go:177]: does this have an effect?

File: cli/cmd/trigger_delivery.go [hunk 2]
```diff
@@ -170,39 +171,36 @@ func doTriggerDelivery(deliveryInputData deliveryStruct) error {
 
 func doTriggerDeliveryPreRunCheck(deliveryInputData deliveryStruct) error {
```
Comment (warber) [inline cli/cmd/trigger_delivery.go:172]: let's rename this to `checkImageAvailable(imgName string)`

File: cli/cmd/trigger_delivery.go [hunk 3]
```diff
@@ -170,39 +171,36 @@ func doTriggerDelivery(deliveryInputData deliveryStruct) error {
...
+	return docker.CheckImageAvailability(image, tag, nil)
 }
 
 func init() {
 	triggerCmd.AddCommand(triggerDeliveryCmd)
 
 	delivery.Project = triggerDeliveryCmd.Flags().StringP("project", "", "",
 		"The project containing the service for which the new artifact will be delivered")
-	triggerDeliveryCmd.MarkFlagRequired("project")
+	_ = triggerDeliveryCmd.MarkFlagRequired("project")
```
Comment (warber) [inline cli/cmd/trigger_delivery.go:186]: remove assignment to blank identifier?

File: cli/cmd/trigger_delivery.go (hunk 1 above)
Comment (bacherfl) [reply cli/cmd/trigger_delivery.go:177]: the `SplitImageName` function checks if there is already a tag defined in the provided image, and defaults it to `latest`, if there is none. Therefore I decided to keep using it

Comment (warber) [reply cli/cmd/trigger_delivery.go:177]: i mean the assignment in line 177

Comment (bacherfl) [reply cli/cmd/trigger_delivery.go]: nope, that did not have an effect. has been removed in the meantime

**Review by warber — APPROVED** [review]

**Pull request merged**